python_modules/libraries/dagster-msteams/dagster_msteams/resources.py

```diff
@@ -1,25 +1,76 @@
-from dagster import Bool, Field, Float, StringSource, resource
+from dagster import ConfigurableResource, resource
+from pydantic import Field
 
 from dagster_msteams.client import TeamsClient
 
 
-@resource(
-    {
-        "hook_url": Field(
-            StringSource,
-            description="""To send messages to MS Teams channel, an incoming webhook has to
-                    be created. The incoming webhook url must be given as a part of the
-                    resource config to the msteams_resource in dagster.
-                    """,
+class MSTeamsResource(ConfigurableResource):
+    """This resource is for connecting to Microsoft Teams.
+
+    The resource object is a `dagster_msteams.TeamsClient`.
+
+    By configuring this resource, you can post messages to MS Teams from any Dagster solid:
+
+    Examples:
+        .. code-block:: python
+
+            import os
+
+            from dagster import op, job, Definitions, EnvVar
+            from dagster_msteams import Card, MSTeamsResource
+
+
+            @op
+            def teams_op(msteams: MSTeamsResource):
+                card = Card()
+                card.add_attachment(text_message="Hello There !!")
+                msteams.get_client().post_message(payload=card.payload)
+
+
+            @job
+            def teams_job():
+                teams_op()
+
+            defs = Definitions(
+                jobs=[teams_job],
+                resources={
+                    "msteams": MSTeamsResource(
+                        hook_url=EnvVar("TEAMS_WEBHOOK_URL")
+                    )
+                }
+            )
+    """
+
+    hook_url: str = Field(
+        default=None,
+        description=(
+            "To send messages to MS Teams channel, an incoming webhook has to be created. The"
+            " incoming webhook url must be given as a part of the resource config to the"
+            " msteams_resource in Dagster."
```

Comment:
```suggestion
            " MSTeamsResource in Dagster."
```

Reply:
not necessary, but it could be helpful to link out to the docs to create an MS teams webhook https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook?tabs=dotnet